### PR TITLE
feat(outlook): generate plugin xml manifest

### PR DESCRIPTION
**Pull request by chillfox**

### CODE REVIEW

File: src/controllers/outlook.cr [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Outlook < ActionController::Base
+  base "/api/staff/v1/outlook"
+
+  get("/manifest.xml") do
+    manifest = OutlookManifest.new(
+      app_domain: "#{App::PLACE_URI}/outlook/",
```
Comment (stakach) [inline src/controllers/outlook.cr:6]: we don't want to use `App::PLACE_URI` this is for internal comms between the containers
we need to use the current `tenant.domain` and only if `tenant.platform == "office365"`

**Review by stakach — CHANGES_REQUESTED** [review]
looks good, just one minor change

File: src/controllers/outlook.cr (hunk 1 above)
Comment (chillfox) [reply src/controllers/outlook.cr:6]: Ok, can we get the tenant domain and platform without auth?
There is a requirement for the xml manifest to be served without auth.

Comment (chillfox) [reply src/controllers/outlook.cr]: I guess we could make the tenant part of the path, or a query param

Comment (stakach) [reply src/controllers/outlook.cr]: it doesn't need auth, just can't use the helper directly

```crystal
domain_host = request.hostname.as(String)
tenant = Tenant.query.find { domain == domain_host }
```

**Review by stakach — APPROVED** [review]
LGTM, let's hold off merging until the XML is finalised
@jeremyw24 to notify of any further changes required

**Comment by stakach** [comment]
@jeremyw24 any additional XML changes now that the office plugin is working?

**Comment by chillfox** [comment]
@stakach 
Since your last review of this the following changes has been made:
- updated xml
- added `outlook_config` json column to tenants for storing config.

**Review by stakach — APPROVED** [review]
Awesome work, looking good!

**Pull request merged**